word_segmentation.py

```diff
@@ -0,0 +1,709 @@
+import numpy as np
+import os
+import icu
+import datetime
+from icu import UnicodeString, BreakIterator, Locale
+import matplotlib.pyplot as plt
+
+import tensorflow as tf
+from keras.models import Sequential
+from keras.layers import LSTM
+from keras.layers import Dense
+from keras.layers import TimeDistributed
+from keras.layers import Bidirectional
+from keras.layers import Embedding
+from keras.layers import Dropout
+
+
+def is_all_english(str):
+    """
+    A very basic function that checks if all elements of str are english characters or common expressions
+    It just checks if the integer assigned to each character is below 130 or not
+    TO DO: This function can be potentially be replace with some ICU function
+    str: input string
+    """
+
+    str = str.replace(" ", "")
+    str = str.replace("|", "")
+    # for k in range(0, 130):
+    #     print(chr(k))
+    for ch in str:
+        if ord(ch) > 130:
+            # print("character = {}, ord = {}".format(ch, ord(ch)))
+            return False
+    return True
+
+
+def sigmoid(x):
+    """
+    Computes the sigmoid function for a scalar
+    x: the scalar
+    """
+    return 1.0/(1.0+np.exp(-x))
+
+
+def analyze_grapheme_clusters(ratios, thrsh):
```

Comment:
This function doesn't return anything; it just prints to the console.  The verb "analyze" is not precise enough.  I would name the function "print_grapheme_clusters" or "plot_grapheme_clusters" to emphasize what the function is actually doing.